array-reduce/main.js

```diff
@@ -0,0 +1,42 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((initialValue, currentValue) => initialValue + currentValue);
+console.log(sum);
+
+const product = numbers.reduce((initialValue, currentValue) => initialValue * currentValue);
+console.log(product);
+
+const balance = account.reduce((initialValue, currentValue) => {
+  let amount = 0;
+  if (currentValue.type === 'deposit') {
+    amount = amount + currentValue.amount;
+  } else {
+    amount = amount - currentValue.amount;
+  }
+  return amount + initialValue;
+}, 0);
+
+console.log(balance);
+
+const composite = traits.reduce((initialValue, currentValue) => {
+  return Object.assign(initialValue, currentValue);
+});
```

Comment:
Be sure to add an initial value (empty object) to copy all of the traits onto. otherwise your first trait is getting all others piled onto it.